### DISCUSSION TITLE
Global CSS stylesheet for NiceGUI

### DISCUSSION THREAD

**Discussion by rosenloecher-it** [zauberzeug/nicegui#1575] [Q&A] [answered]
Is there a way to provide a global CSS style-sheet?

I want to separate code and design. So, is there a way to define my own CSS classes (in a separate CSS style-sheet file) and assign these classes to my elements. I would like to avoid assigning concrete CSS style instructions on every element, e.g. `ui.label('label ').style('color: ...`.

Also I want to override predefined CSS classes.

**Comment by v479038280** [comment]
You can refrence the nicegui site ,in the static folder there is a style.css.

**Reply by superlou** [reply]
How do you get the app to reference the `static/style.css` file? I was manually doing it in my app (https://github.com/superlou/nice-datasheet/blob/main/nicesheet/sheet.py#L43), but is this a built-in feature already?

**Reply by v479038280** [reply]
You could also use ` ui.add_head_html((Path(__file__).parent / 'website' / 'static' / 'header.html').read_text())`

**Reply by rosenloecher-it** [reply, marked as answer]
These answers and from [stackoverflow](https://stackoverflow.com/questions/77043716/global-css-stylesheet-for-nicegui/) lead me to 2 solutions:

My favourite:
```python
# style-sheet file is: Path(__file__).parent / "static" / "styles.css"
app.add_static_files('/static', 'static')
ui.add_head_html('<link rel="stylesheet" type="text/css" href="/static/styles.css">')
```
or:
```python
ui.add_head_html("<style>" + open(Path(__file__).parent / "static" / "styles.css").read() + "</style>")
```
Thanks.
